### PR TITLE
Add documentation - Develop on Kuzzle - External Plugins

### PR DESCRIPTION
**Pull request by Aschen**
https://deploy-preview-1844--doc-kuzzle.netlify.app/guides/develop-on-kuzzle/4-external-plugins/

### CODE REVIEW

File: doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md [hunk 1]
```diff
@@ -8,3 +8,64 @@ order: 400
 
 # External Plugins
 
+Kuzzle allows to add **whole set of features via plugins**.
```
Comment (xbill82) [inline doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md:11]:
```suggestion
You can **extend Kuzzle's features** via plugins.
```

File: doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md [hunk 2]
```diff
@@ -8,3 +8,64 @@ order: 400
...
+
+Plugins are **intended to be reused** between several applications by proposing generic functionalities.
```
Comment (xbill82) [inline doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md:13]:
```suggestion
Plugins are **intended to be reused** between several applications by proposing generic features.
```

File: doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md [hunk 3]
```diff
@@ -8,3 +8,64 @@ order: 400
...
+
+There are 3 categories of functionalities for plugins:
```
Comment (xbill82) [inline doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md:15]:
```suggestion
Plugins allow you to achieve three main goals:
```

**Review by xbill82 — APPROVED** [review]

File: doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md [hunk 4]
```diff
@@ -8,3 +8,64 @@ order: 400
...
+You can **extend Kuzzle's features** via plugins.
+
+Plugins are **intended to be reused** between several applications by proposing generic features.
+
+Plugins allow you to achieve three main goals:
+ - extension and modification of the API
+ - integration of an authentication strategy
+ - integration of a network protocol
+
+These feature categories are not exclusive and although it is not recommended, a plugin can embed features belonging to the 3 categories.
```
Comment (scottinet) [inline doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md:20]: nope: plugins and protocols are different objects, with separate directories

File: doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md [hunk 5]
```diff
@@ -8,3 +8,64 @@ order: 400
...
+
+Plugins are classes that must implement the [Plugin](/core/2/some-link) interface.  
+
+They can create new controllers or register pipes and hooks like applications.
+
+::: info
+The controllers registered by the plugins are prefixed by the plugin name in order to avoid any conflict.
+:::
+
+In addition to the standard features available in an application, plugins can also integrate [authentication strategies](/core/2/some-link) and [network protocols](/core/2/some-link).
```
Comment (scottinet) [inline doc/2/guides/develop-on-kuzzle/4-external-plugins/index.md:32]: nope, plugins can't add new network protocols, only protocols can

```suggestion
In addition to the standard features available in an application, plugins can also integrate [authentication strategies](/core/2/some-link) and [network protocols](/core/2/some-link).
```

**Review by scottinet — CHANGES_REQUESTED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

**Review by xbill82 — APPROVED** [review]

**Pull request merged**